routes/v1.js

```diff
@@ -384,7 +386,7 @@ api.declare({
 
 /** Define tasks */
 api.declare({
-  method:     'post',
+  method:     'put',
```

Comment:
I don't like putting `state` into the task definition...
Also using `PATCH` on `/task/<taskId>` for scheduleTask... then what about cancelTask, rerunTask... Should we then extend our API reference format to support methods that specify a hardcoded payload and have the same URL pattern...

We are not truly REST-compliant... And this is a problem through-out all our APIs, we are somewhat REST-like, sorry... Not pure-rest.


-----
@petemoore, So I noticed how badly we violated REST principals a while ago. I saw no easy way to fix it... and after thinking about it I can't find any value from fixing this. Other than it would be most REST-like.

To do REST correctly we would however have to make self-describing resources and paths... Such that navigating to queue.taskcluster.net/v1/ lists all the paths and explains what they do..
And if the `accept` header is `text/html` we should render replies to HTML before returning them...

All in all I see no value in doing this right. It would be awesome to do right for purity, clarity and  I'm sure it would solve problems down the road. But the hassle of doing would be huge.

IMO, doing it right would require some serious abstractions so we can't do the wrong thing. No such API abstractions exists now... As far as I know... So I think we should file this **under some future**, just like we did with the idea of doing versioning of everything correctly. It would require abstractions and engineering to do it right, things we're not willing to invest in now.

Also I seriously doubt super purity has much value. Azure and AWS aren't always pure REST either (well, it depends for sure).